pkg/controller/rsm2/instance_util.go

```diff
@@ -254,6 +254,8 @@ func buildInstanceByTemplate(name string, template *instanceTemplateExt, parent
 		AddAnnotationsInMap(template.Annotations).
 		AddLabelsInMap(template.Labels).
 		AddControllerRevisionHashLabel(revision).
+		// for backward compatibility to support pod-service binding
+		AddLabelsInMap(map[string]string{constant.StatefulSetPodNameLabelKey: name}).
```

Comment:
Use a new key, such as RSMPodNamLabelKey, instead of StatefulSetPodNameLabelKey? Since the new Component Service API and Vars reference are not compatible with the previous one, so there is no need for compatibility here

Reply:
change the label key to: apps.kubeblocks.io/pod-name